app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -371,6 +371,15 @@ else if(resultCode == RESULT_OK){
                     return;
                 } else if(resultCode == RESULT_OK) {
                     if (!intent.getBooleanExtra(LoginActivity.ALREADY_LOGGED_IN, false)) {
+                        CommCareSession session = CommCareApplication._().getCurrentSession();
+                        if (session.getCommand() != null) {
+                            // restore the session state if there is a command.
+                            // For debugging and occurs when a serialized
+                            // session is stored upon login
+                            sessionNavigator.startNextSessionStep();
+                            return;
+                        }
```

Comment:
I'm not sure... You're essentially trying to check whether the session has any info in it yet, right? Another option would be to just set a flag in the case when you restore the session in CommCareApplication.bindUserSessionService(), and then we know for sure that it's correct. Otherwise, would want to double check with @ctsims that this is the right way of doing it